interpreter/runtime/event.ml

```diff
@@ -0,0 +1,10 @@
+open Types
+
+type event = {ty : func_type}
```

Comment:
Hm, it's a bit odd that this is func_type not event_type. I can see why, but it would break down once an event type has additional attributes.

The underlying problem is that there are two notions of event_type: the syntactic one (with a typeidx) and the semantic one (with the function type inline). Depending on context, we want one or the other. This is not a situation we've had before, so I don't have any great suggestion right now. We can resolve this later.

Reply:
Ack. Regarding the syntactic vs semantic `event_type`: I don't see how this is too different from functions, where the type can also be represented directly or through the type ID? It makes sense to me that if one uses the "semantic" encoding, the other does too.

Reply:
The difference is the extra indirection. For function types, the type section is just mapping a typeidx to a func type, and the functype itself is a closed type description. Once looked up, we can use that directly. But for event types, the definition itself refers to another typeidx, so isn't self-contained. This becomes clearer if we assume that an event type has additional ingredients, say, a hypothetical event_mode (as per the reserved zero byte). Then for the syntax, we'd need 
```event_type = EventType of typeidx * event_mode```
But but when e.g. defining extern_type, that index is meaningless (because it's used outside any module) and you want `ExternEventType of event_type'` where
```event_type' = EventType' of func_type * event_mode```
FWIW, a similar problem also arises with typed reference types, so the syntactic vs semantic distinction becomes rather ubiquitous.

Reply:
I agree, and for this reason I think types.ml should contain the resolved type.
We already have the typeidx in the AST:
```
type event = event' Source.phrase
and event' =
{
  etype : var;
}
```
which makes sense for encoding, decoding and pretty-printing. For type checking, imports/exports and everything else, the resolved type is easier to work with. WDYT about simply reverting back to `type event_type = EventType of func_type`?

Reply:
Actually, both `etype` and `EventImport` ought to carry an `event_type`, not a `var`. Otherwise, they in turn don't factorise correctly, e.g., when event types were to consist of more than just a var.

I'm afraid we can't hack around the two facets, they're inherent. It's relatively localised here, but becomes ubiquitous with concrete reference types. In the function references proposal I currently resolve it by actually distinguishing syntactic and semantic types. They differ by how variables are represented: for syntactic types it's just an index, with semantic types they become a pointer to the actual definition.

But I suggest to not worry about it for now. I want to play with streamlining this a bit and can fix it later. This needs a representation that cleanly carries over to the paper spec.